internal/driver/interactive_test.go

```diff
@@ -40,62 +40,53 @@ func TestShell(t *testing.T) {
+		{"Random interleave of independent scripts with shortcuts 2", scScript2, shortcuts2, "", 0, false},
+		{"Group with invalid vale", []string{"cumulative=this"}, pprofShortcuts, `unrecognized value for cumulative: "this". Use one of cum, flat`, 1, false},
+		{"No special value provided for the option", []string{"sample_index"}, pprofShortcuts, `please input a value, e.g. sample_index=<val>`, 1, false},
+		{"No string value provided for the option", []string{"focus="}, pprofShortcuts, `please input a value, e.g. focus=<val>`, 1, false},
+		{"No float value provided for the option", []string{"divide_by"}, pprofShortcuts, `please input a value, e.g. divide_by=<val>`, 1, false},
+		{"Helpful input format reminder", []string{"sample_index 0"}, pprofShortcuts, `did you mean: sample_index=0`, 1, false},
+		{"Verify propagation of IO errors", []string{"**error**"}, pprofShortcuts, "", 0, false},
 	}
 
-	// Random interleave of independent scripts with shortcuts
-	pprofVariables = testVariables(savedVariables)
-	var scScript []string
-	pprofShortcuts, scScript = makeShortcuts(interleave(script, 2), 1)
-	o.UI = newUI(t, scScript)
-	if err := interactive(p, o); err != nil {
-		t.Error("first shortcut attempt:", err)
-	}
+	o := setDefaults(&plugin.Options{HTTPTransport: transport.New(nil)})
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			pprofVariables = testVariables(savedVariables)
+			pprofShortcuts = tc.shortcuts
+			ui := &proftest.TestUI{
+				T:       t,
+				Input:   tc.input,
+				AllowRx: tc.allowRx,
+			}
+			o.UI = ui
 
-	// Random interleave of independent scripts with shortcuts
-	pprofVariables = testVariables(savedVariables)
-	pprofShortcuts, scScript = makeShortcuts(interleave(script, 1), 2)
-	o.UI = newUI(t, scScript)
-	if err := interactive(p, o); err != nil {
-		t.Error("second shortcut attempt:", err)
-	}
+			err := interactive(p, o)
+			if (tc.propagateError && err == nil) || (tc.propagateError && err != nil) {
+				t.Errorf("%v: %v", tc.name, err)
+			}
 
-	// Group with invalid value
-	pprofVariables = testVariables(savedVariables)
-	ui := &proftest.TestUI{
-		T:       t,
-		Input:   []string{"cumulative=this"},
-		AllowRx: `unrecognized value for cumulative: "this". Use one of cum, flat`,
-	}
-	o.UI = ui
-	if err := interactive(p, o); err != nil {
-		t.Error("invalid group value:", err)
-	}
-	// Confirm error message written out once.
-	if ui.NumAllowRxMatches != 1 {
-		t.Errorf("want error message to be printed 1 time, got %v", ui.NumAllowRxMatches)
-	}
-	// Verify propagation of IO errors
-	pprofVariables = testVariables(savedVariables)
-	o.UI = newUI(t, []string{"**error**"})
-	if err := interactive(p, o); err == nil {
-		t.Error("expected IO error, got nil")
+			// Confirm error message written out once.
+			if tc.numAllowRxMatches != ui.NumAllowRxMatches {
+				t.Errorf("want error message to be printed %v time(s), got %v",
```

Comment:
Use `%d` instead of `%v`?
(I know this line already existed, but nice to be able to make it a bit more idiomatic.)

Reply:
Done.